crates/volta-core/src/monitor.rs

```diff
@@ -1,58 +1,79 @@
 use std::io::Write;
+use std::path::PathBuf;
 use std::process::{Child, Stdio};
 
 use log::debug;
+use tempfile::NamedTempFile;
 
 use crate::command::create_command;
 use crate::event::Event;
 
-pub struct Monitor {
-    monitor_process: Option<Child>,
-}
-
-impl Monitor {
-    /// Returns the current monitor.
-    pub fn new(command: &str) -> Monitor {
-        Monitor {
-            monitor_process: spawn_process(command),
+/// Send event to the spawned command process
+// if hook command is not configured, this is not called
+pub fn send_events(command: &str, events: &[Event]) {
+    match serde_json::to_string_pretty(&events) {
+        Ok(events_json) => {
+            if let Some(tempfile_path) = write_events_file(events_json.clone()) {
```

Comment:
May not be a major issue, but since File I/O can be slow, is there a way we can gate the `write_events_file` piece behind an environment variable for use in the tests? The public API for the event hook is still that the data will be written to `stdin`, so we can have the extra bit for use in the tests while keeping the performance of the existing solution.

Reply:
Good point - updated that in 9c857a289a511a202a5ccfc20f06212d527f5194